library/src/main/java/org/apache/apex/malhar/lib/utils/serde/SerToSerializeBuffer.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.utils.serde;
+
+import com.datatorrent.netlet.util.Slice;
+
+public interface SerToSerializeBuffer<T> extends Serde<T, Slice>, ResetableWindowListener
```

Comment:
A lot of unit test code and serde only depended on Serde right now. If I changed SpillableDS depended on SerToSerializeBuffer or merge SerToSerializeBuffer into Serde. a lot of code would be changed. Due this PR already pretty large. Maybe use another task to merge them.